sympy/solvers/tests/test_ode.py

```diff
@@ -3076,66 +3081,84 @@ def test_issue_7081():
 
 @slow
 def test_2nd_power_series_ordinary():
-    # FIXME: Maybe there should be a way to check series solutions
-    # checkodesol doesn't work with them.
     C1, C2 = symbols("C1 C2")
+
     eq = f(x).diff(x, 2) - x*f(x)
     assert classify_ode(eq) == ('2nd_power_series_ordinary',)
-    assert dsolve(eq) == Eq(f(x),
-        C2*(x**3/6 + 1) + C1*x*(x**3/12 + 1) + O(x**6))
-    assert dsolve(eq, x0=-2) == Eq(f(x),
-        C2*((x + 2)**4/6 + (x + 2)**3/6 - (x + 2)**2 + 1)
+    sol = Eq(f(x), C2*(x**3/6 + 1) + C1*x*(x**3/12 + 1) + O(x**6))
+    assert dsolve(eq) == sol
+    assert checkodesol(eq, sol) == (True, 0)
+
+    sol = Eq(f(x), C2*((x + 2)**4/6 + (x + 2)**3/6 - (x + 2)**2 + 1)
         + C1*(x + (x + 2)**4/12 - (x + 2)**3/3 + S(2))
         + O(x**6))
-    assert dsolve(eq, n=2) == Eq(f(x), C2*x + C1 + O(x**2))
+    assert dsolve(eq, x0=-2) == sol
+    # FIXME: Solution should be O((x+2)**6)
```

Comment:
`O((x+2)**6)` has a different meaning. So we can't get from the solution and also O(x**6) is the correct way of representing the solution. 
```
In [9]: Order((x+2)**6)
Out[9]: O(1)

```

For the` checkodesol` We can pass the argument `x0` and then we can use it for your work.

Reply:
What I meant is:
```julia
In [1]: O((x+2)**6, (x, -2))                                                                                                      
Out[1]: 
 ⎛       6        ⎞
O⎝(x + 2) ; x → -2⎠
```

Reply:
Making `dsolve` return that fails because `constantsimp` calls `cse` which absorbs all the terms #17587